x/stakeibc/keeper/reward_converter.go

```diff
@@ -51,6 +57,67 @@ type ForwardMetadata struct {
 // and the normal staking and distribution flow will continue from there.
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+// Checks if a rebate exists for the given host zone
+// If so, splits the rewardCoin into a portion for the rebate and a portion that should get reinvested
+// Returns the rebate amount, the stride fee amount, and the reinvestment amount
+//
+// The rebate percentage is determined by: (% of total TVL contributed by commuity pool) * (rebate percentage)
+//
+// E.g. Community pool liquid staked 1M, TVL is 10M, rebate is 50%
+// Total rewards this epoch are 1000, and the stride fee is 10%
+// => Then the rebate is 1000 rewards * 10% stride fee * (1M / 10M) * 50% rebate = 5
+func (k Keeper) CheckForCommunityPoolRebate(
+	ctx sdk.Context,
+	chainId string,
+	rewardAmount sdkmath.Int,
+) (feeInfo FeeInfo, err error) {
+	hostZone, err := k.GetActiveHostZone(ctx, chainId)
+	if err != nil {
+		return FeeInfo{}, err
+	}
+
+	// Get the fee percentage from params
+	strideFee := sdk.NewIntFromUint64(k.GetParams(ctx).StrideCommission)
+	strideFeePct := sdk.NewDecFromInt(strideFee).Quo(sdk.NewDec(100))
+
+	// Get the total fee amount from the fee percentage
+	totalFeesAmount := sdk.NewDecFromInt(rewardAmount).Mul(strideFeePct).TruncateInt()
+	reinvestAmount := rewardAmount.Sub(totalFeesAmount)
+
+	// Get the rebate info from the host zone if applicable
+	// If there's no rebate, return 0 rebate and reinvest the full reward
+	rebateInfo, chainHasRebate := hostZone.SafelyGetCommunityPoolRebate()
+	if !chainHasRebate {
+		feeInfo = FeeInfo{
+			RebateAmount:    sdkmath.ZeroInt(),
+			StrideFeeAmount: totalFeesAmount,
+			ReinvestAmount:  reinvestAmount,
+		}
+		return feeInfo, nil
+	}
+
+	// It shouldn't be possible to have 0 delegations (since there are rewards and there was a community pool stake)
+	// This will also prevent a division by 0 error
+	if hostZone.TotalDelegations.IsZero() {
+		return feeInfo, errorsmod.Wrapf(types.ErrDivisionByZero,
+			"unable to calculate rebate amount for %s since total delegations are 0", chainId)
+	}
+
+	// The rebate amount is determined by the contribution of the community pool stake towards the total TVL,
+	// multiplied by the rebate fee percentage
+	contributePercentage := sdk.NewDecFromInt(rebateInfo.LiquidStakeAmount).Quo(sdk.NewDecFromInt(hostZone.TotalDelegations))
+	rebateAmount := sdk.NewDecFromInt(totalFeesAmount).Mul(contributePercentage).Mul(rebateInfo.RebatePercentage).TruncateInt()
+	strideFeeAmount := totalFeesAmount.Sub(rebateAmount)
```

Comment:
Should we add some sanity checks in here e.g.
- strideFeeAmount > 0 (should it just be a uint?)
- 0 <= contributePercentage <= 1
- rebateAmount >= 0

Reply:
duplicate of above, responded there